### PR TITLE
add support for wildcard filename

### PR DESCRIPTION
**Pull request by IanSmith123**

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -786,3 +791,38 @@ func (t *Tag) validateAndReset(prefix string, start int) error {
 	}
 	return nil
 }
+
+// TODO(yuyu): discuss about the config schema for import by directory
+func (config *YAMLConfig) expandDirectoryToFiles(dir string) error {
```
Comment (yixinglu) [inline pkg/config/config.go:796]:
```go
func (config *YAMLConfig) expandDirectoryToFiles(dir string) (err error) {
...
```

File: pkg/config/config.go [hunk 2]
```diff
@@ -786,3 +791,38 @@ func (t *Tag) validateAndReset(prefix string, start int) error {
...
+
+	var err error
```
Comment (yixinglu) [inline pkg/config/config.go:798]: this declaration could be deleted.

File: pkg/config/config.go [hunk 3]
```diff
@@ -786,3 +791,38 @@ func (t *Tag) validateAndReset(prefix string, start int) error {
...
+	var newFiles []*File
+
+	for i := range config.Files {
+		if !base.HasHttpPrefix(*config.Files[i].Path) {
+			// treat all string as glob pattern
+			// query by wildcard
+			files, err := filepath.Glob(filepath.Join(dir, *config.Files[i].Path))
+			if err != nil || len(files) == 0 {
+				err = errors.New(fmt.Sprintf("error string: %s", *config.Files[i].Path))
+				logger.Errorf("%s", err)
```
Comment (yixinglu) [inline pkg/config/config.go:809]: `return err`

File: pkg/config/config.go [hunk 4]
```diff
@@ -786,3 +791,38 @@ func (t *Tag) validateAndReset(prefix string, start int) error {
...
+			// maybe I missed the usage of pointer :(
+			logger.Infof("query file from pattern: %s, result: %v", *config.Files[i].Path, files)
+			for j := range files {
+				eachConf := config.Files[i]
+				eachConf.Path = &files[j]
+				newFiles = append(newFiles, eachConf)
+			}
+		} else {
+			newFiles = append(newFiles, config.Files[i])
+
```
Comment (yixinglu) [inline pkg/config/config.go:821]: cleanup blank line

File: pkg/config/config.go [hunk 5]
```diff
@@ -786,3 +791,38 @@ func (t *Tag) validateAndReset(prefix string, start int) error {
...
+		}
+
```
Comment (yixinglu) [inline pkg/config/config.go:823]: ditto

**Review by yixinglu — COMMENTED** [review]
Good Job!

**Comment by IanSmith123** [comment]
Thanks for your review, I have made changes based on your suggestions.

File: pkg/config/config.go [hunk 6]
```diff
@@ -183,6 +183,11 @@ func (config *YAMLConfig) ValidateAndReset(dir string) error {
 		return errors.New("There is no files in configuration")
 	}
 
+	//TODO(yuyu): check each item in config.Files
+	// if item is a directory, iter this directory and replace this directory config section by filename config section
+	if err := config.expandDirectoryToFiles(dir); err != nil {
+		logger.Errorf("%s", err)
```
Comment (yixinglu) [inline pkg/config/config.go:189]: return err

**Comment by IanSmith123** [comment]
Thanks for your review, I have made changes based on your suggestions again.  😃

**Review by yixinglu — APPROVED** [review]
Excellent!!

**Pull request merged**